packages/fast-colors/src/component-state-color-palette.ts

```diff
@@ -0,0 +1,165 @@
+import { ColorRGBA64 } from "./color-rgba-64";
+import { ColorPalette } from "./color-palette";
+import { hslToRGB, rgbToHSL, rgbToLuminanceLinear } from "./color-converters";
+import { ColorScale, ColorScaleStop } from "./color-scale";
+import { ColorHSL } from "./color-hsl";
+import { ColorInterpolationSpace } from "./color-interpolation";
+import { parseColorHexRGB } from "./parse-color";
+
+export interface ComponentStateColorPaletteConfig {
+    baseColor?: ColorRGBA64;
+    steps?: number;
+}
+
+export class ComponentStateColorPalette {
+    public static readonly defaultPaletteConfig: ComponentStateColorPaletteConfig = {
+        baseColor: parseColorHexRGB("#808080"),
```

Comment:
same question here - can we not just use `.5` middle values and avoid the parsing step?

Reply:
Response above.